sdks/go/test/integration/io/fhirio/fhirio_test.go

```diff
@@ -0,0 +1,198 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fhirio
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"math/big"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/fhirio"
+	_ "github.com/apache/beam/sdks/v2/go/pkg/beam/runners/dataflow"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+	"github.com/apache/beam/sdks/v2/go/test/integration"
+	"google.golang.org/api/healthcare/v1"
+	"google.golang.org/api/option"
+)
+
+const (
+	datasetPathFmt = "projects/%s/locations/%s/datasets/apache-beam-integration-testing"
+	testDataDir    = "../../../../data/fhir_bundles/"
+)
+
+var (
+	gcpProject             string
+	gcpRegion              string
+	storeService           *healthcare.ProjectsLocationsDatasetsFhirStoresFhirService
+	storeManagementService *healthcare.ProjectsLocationsDatasetsFhirStoresService
+)
+
+func checkFlags(t *testing.T) {
+	if gcpProject == "" || gcpRegion == "" {
+		t.Skip("GCP flags not provided.")
+	}
+}
+
+// Sets up a test fhir store by creating and populating data to it for testing
+// purposes. It returns the name of the created store path, a slice of the
+// resource paths to be used in tests, and a function to teardown what has been
+// set up.
+func setupFhirStore(t *testing.T) (string, []string, func()) {
+	t.Helper()
+	if storeService == nil || storeManagementService == nil {
+		t.Fatal("Healthcare Services were not initialized")
+	}
+
+	healthcareDataset := fmt.Sprintf(datasetPathFmt, gcpProject, gcpRegion)
```

Comment:
These tests are enabled in direct runner, but this requires a gcp project and gcp region to be specified, which they normally aren't with the direct runner. My first instinct is that these tests wouldn't work when run on the direct runner with our current integration test framework. Have you confirmed whether it does or not?

Reply:
Ok I just checked, the direct runner integration tests aren't handled by any of our postcommits which might be an oversight. The precommit does technically test this directory, but it's just doing a plain "go test ./..." command, so it doesn't provide any of the necessary flags (it's technically just aimed at unit tests).

Reply:
Last comment, but I think what you should aim at supporting is the portable runner, which is basically the more fully featured "direct" runner. And you might need to pipe a flag directly to the test by [adding something here](https://github.com/apache/beam/blob/master/sdks/go/test/build.gradle#L147).

Reply:
Resolving this since as per chat discussion.